internal/services/network/custom_ip_prefix_resource.go

```diff
@@ -0,0 +1,550 @@
+				return err
+			}
+		}
+
+		if currentState == desiredState {
+			return nil
+		}
+
+		for state, path := range plan {
+			if currentState == state || transitioningStatesFor(state).contains(currentState) {
+				if err := r.waitForCommissionedState(ctx, id, transitioningStatesFor(state).strings(), []string{string(state)}); err != nil {
+					return err
+				}
+
+				for _, steppingState := range path {
+					var noInternetAdvertise *bool
+					if steppingState == network.CommissionedStateCommissioning {
+						switch desiredState {
+						case network.CommissionedStateCommissioned:
+							noInternetAdvertise = pointer.To(false)
+						case network.CommissionedStateCommissionedNoInternetAdvertise:
+							noInternetAdvertise = pointer.To(true)
+						}
+					}
+
+					if err := r.setCommissionedState(ctx, id, steppingState, noInternetAdvertise); err != nil {
+						return err
+					}
+
+					if err := r.waitForCommissionedState(ctx, id, []string{string(steppingState)}, []string{string(finalStateFor(steppingState))}); err != nil {
+						return err
+					}
+				}
+
+				return r.waitForCommissionedState(ctx, id, transitioningStatesFor(desiredState).strings(), []string{string(desiredState)})
+			}
+		}
+	} else {
+		return fmt.Errorf("unsupported state %q", desiredState)
+	}
+
+	return nil
+}
+
+func (r CustomIpPrefixResource) setCommissionedState(ctx context.Context, id parse.CustomIpPrefixId, desiredState network.CommissionedState, noInternetAdvertise *bool) error {
+	existing, err := r.client.Get(ctx, id.ResourceGroup, id.CustomIpPrefixeName, "")
+	if err != nil {
+		return fmt.Errorf("retrieving existing %s: %+v", id, err)
+	}
+	if existing.CustomIPPrefixPropertiesFormat == nil {
+		return fmt.Errorf("retrieving existing %s: `properties` was nil", id)
+	}
+
+	existing.CustomIPPrefixPropertiesFormat.CommissionedState = desiredState
+	existing.CustomIPPrefixPropertiesFormat.NoInternetAdvertise = noInternetAdvertise
+
+	log.Printf("[DEBUG] Updating the CommissionedState field to %q for %s..", string(desiredState), id)
+	future, err := r.client.CreateOrUpdate(ctx, id.ResourceGroup, id.CustomIpPrefixeName, existing)
+	if err != nil {
+		return fmt.Errorf("updating CommissionedState to %q for %s: %+v", string(desiredState), id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, r.client.Client); err != nil {
+		return fmt.Errorf("waiting for the update of CommissionedState to %q for %s: %+v", string(desiredState), id, err)
+	}
+
+	return nil
+}
+
+func (r CustomIpPrefixResource) waitForCommissionedState(ctx context.Context, id parse.CustomIpPrefixId, pendingStates, targetStates []string) error {
+	log.Printf("[DEBUG] Polling for the CommissionedState field for %s..", id)
+	timeout, _ := ctx.Deadline()
+
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending:    pendingStates,
+		Target:     targetStates,
+		Refresh:    r.commissionedStateRefreshFunc(ctx, id),
+		MinTimeout: 1 * time.Minute,
```

Comment:
As above, should we consider using `PollInterval` here?

Reply:
Yeah makes sense for this check, as it can take 4-17 hours to complete (!!)